release/pkg/manager/calico/manager.go

```diff
@@ -901,6 +902,11 @@ func (r *CalicoManager) generateManifests() error {
 	env := os.Environ()
 	env = append(env, fmt.Sprintf("CALICO_VERSION=%s", r.calicoVersion))
 	env = append(env, fmt.Sprintf("OPERATOR_VERSION=%s", r.operatorVersion))
+	env = append(env, fmt.Sprintf("OPERATOR_REGISTRY=%s", r.operatorRegistry))
+	env = append(env, fmt.Sprintf("OPERATOR_IMAGE=%s", r.operatorImage))
```

Comment:
I need to think about this a little bit more. I'm not very comfortable with the complexity this seems to be introducing. It's a lot more difficult to understand exactly where the image / registry used in a proper release is meant to come from.

I think this falls into the same category of duplicating what `values.yaml` is doing. I think we want `charts/tigera-operator/values.yaml` to be the source of truth for releases and hashreleases, and the CLI flags passed to the release tool are an optional override for dev releases.

Reply:
> It's a lot more difficult to understand exactly where the image / registry used in a proper release is meant to come from
this section is only used for hashreleases here: https://github.com/projectcalico/calico/blob/master/release/pkg/manager/calico/manager.go#L296-L316

> I think this falls into the same category of duplicating what values.yaml is doing
This section is always using values from the `release/` flags defaults. If we want to define SOT for variables, i reckon it will have to start there which IMO will go beyond the scope of this PR.

I will like us to have clear delineation for where SOT comes from for a lot of the variables used by the release tool esp WRT default values for flags that the tool consumes

Reply:
Yeah, I think it might have been a bug that I missed in previous reviews the source of truth was moved to `release/` instead of using the `values.yaml` file. 

To be clear, I'm not asking you to fix in this PR, but right not we're in an uncomfortable place where we have multiple sources of truth for different purposes (per-PR CI uses values.yaml, hashrelease and release use the `release/` package values).